lib/ui/surveys/survey_view.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter_ic/gen/assets.gen.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
+import 'package:survey_flutter_ic/widget/circle_button.dart';
+
+class SurveyView extends ConsumerStatefulWidget {
+  final List<SurveyModel> surveys;
+  final Function(int) onPageChanged;
+  final Function(SurveyModel) onSurveySelected;
+
+  const SurveyView(
+      {Key? key,
+      required this.surveys,
+      required this.onPageChanged,
+      required this.onSurveySelected})
+      : super(key: key);
+
+  @override
+  ConsumerState<SurveyView> createState() => _SurveyViewState();
+}
+
+class _SurveyViewState extends ConsumerState<SurveyView> {
+  final PageController _pageController = PageController(initialPage: 0);
+
+  @override
+  void dispose() {
```

Comment:
@hoangnguyen92dn We usually put the `dispose` at the bottom of the widget for readability 🙏